docs/source/development/areas/integrate_pipeline_framework.rst

```diff
@@ -1,4 +1,171 @@
 Integrate a New Pipeline Framework
 ==================================
 
-[COMING SOON]
+.. include:: ../../snippets/slack_support.rstinc
+
+One major feature of Lineapy is the ability to create pipelines that run various pipeline orchestration frameworks.
+We believe that supporting pipeline frameworks to meet data engineers where they are with their current stacks provides the best user experience for reducing manual work of pipeline writing.
+
+Read more about the concept of a pipeline :ref:`here <pipeline_concept>` and Lineapy Pipelines :ref:`here <pipelines>`.
+
+In this tutorial we will go through how to write an new pipeline integration. We'll first go through important concepts before diving into concrete steps. If you want to dive right into the concrete steps, skip ahead to the :ref:`checklist <pipeline_integration_checklist>`.
+
+
+This tutorial will use airflow as an illustrative example, but there may not be a one-size-fits-all approach for all integrations so please adapt the content in this guide to suite your individual needs.
+
+The module file
+---------------
+
+The module file contains all the user code from the original notebook needed to run a pipeline. The module file is one of a few files that Lineapy generates to create pipelines, an overview of which can be found here :ref:`here <iris_pipeline_module>`.
+
+The module file contains code that is already cleaned and refactored based on parameters passed to a ``to_pipeline`` call and is presented as a python module file. 
+
+This file can and should be used by integrations. The key for new integrations is to figure out how to call these functions within the constraints of their framework. Fortunately, we provide another abstraction to help make this easy. 
+
+
+What is a TaskDefinition?
+--------------------------
+
+TaskDefinition is the abstraction that Lineapy provides integration writers that provides all the information needed to call the functions in the module file.
+Our goal is to have TaskDefinition provide modular blocks that can be used by multiple frameworks with some help from Jinja templating to provide framework specific syntax. 
+
+.. note::
+    The key idea behind a TaskDefinition is that we believe that a task should correspond to whatever the "unit of execution" is for your framework. 
+    For example, in Airflow, this would be the concept of an Operator.
+
+For the latest attributes in TaskDefinition, please refer to our API reference: :class:`lineapy.plugins.task.TaskDefinition`.
+
+.. note::
+    If you feel a code block is missing from TaskDefinition create a separate PR to add it along with an example of why your framework needs it.
+
+Getting TaskDefinitions
+------------------------
+
+Functions to get TaskDefinitions should be located in ``lineapy/plugins/taskgen.py``.
```

Comment:
I worry that the reader may take "TaskDefinitions" (plural) to be a separate construct different from "TaskDefinition". To avoid potential confusion, can we be extra clear and use the code-style for referring TaskDefinition? That is, for the plural, it will be styled as `TaskDefinition`s.

Reply:
Btw, you can apply the same code styling in section headers as well.

Reply:
This looks really funny, I can't get it to render without a space. 

\``TaskDefinition``s 
renders as 
<img width="307" alt="image" src="https://user-images.githubusercontent.com/7387520/203111999-bbcc41ea-37f5-47aa-98e8-94b0dfe93cff.png">

but 

\``TaskDefinition`` s 
renders as 
<img width="248" alt="image" src="https://user-images.githubusercontent.com/7387520/203112104-a07aa525-0f6f-4411-ae16-5f7fe8bf4536.png">
with a space between the "definition" and the "s". 


I'd rather keep things as is if we can't get this to render cleanly. Also can't figure out how to render code style in a link 
<img width="219" alt="image" src="https://user-images.githubusercontent.com/7387520/203112840-dff91aee-e54a-4371-b70a-7a35f860f7e3.png">

Reply:
I see, yeah, I just checked and it seems like what I suggested does not render proper in RST (thought it would be the same as in MD). Okay, let's keep it as is and revisit this if the reader brings up the issue (if at all).